modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -742,6 +756,53 @@ NSFVAction::act()
 
     addBoundaryPostprocessors();
   }
+
+  if (getParam<bool>("initialize_variables_from_mesh_file"))
+  {
+    if (getParam<bool>("pin_pressure"))
+      mooseError("does it work to load scalar variable from exodus??");
```

Comment:
yes it does
see this input for example
https://github.com/idaholab/virtual_test_bed/blob/Newlogo/msr/msfr/steady/run_ns.i